pkg/objstore/objstore.go

```diff
@@ -137,12 +140,36 @@ func TryToGetSize(r io.Reader) (int64, error) {
 		return fileInfo.Size(), nil
 	case *bytes.Buffer:
 		return int64(f.Len()), nil
+	case *bytes.Reader:
+		// Returns length of unread data only.
+		return int64(f.Len()), nil
 	case *strings.Reader:
 		return f.Size(), nil
+	case Sizer:
+		return f.Size()
 	}
 	return 0, errors.Errorf("unsupported type of io.Reader: %T", r)
 }
 
+// Sizer can return size of data.
+type Sizer interface {
+	// Size returns length of the data, or error, if it is not available.
+	// It is best to call Size before calling other methods on the type, as some io.Reader implementations may only return
```

Comment:
I get this comment but now that `Sizer` is a dedicated interface this comment is harder to understand.

Reply:
What's your suggestion? Do you want me to revert back to `ReaderWithSize` or improve comment (suggestions welcome).

Reply:
Sorry for being unclear. I was just suggesting to improve the comment.